### DISCUSSION TITLE
Building firmware of crazyflie after modification

### DISCUSSION THREAD

**Discussion by HazarBoughanmi** [bitcraze/discussions#85] [Q&A]
Hi all!

I try to add some files to the crazyflie fitmware. 
To compile, I use these commands: **$ make cf2_defconfig & $ make -j 12** but I got errors (undefined references to some functions I added) so I need to build the firmware from scratch, and let the linker find all include addresses.

So my question : how to build the firmware after modifying it ?

Thank you.

**Comment by krichardsson** [comment]
If you add new .c files you have to add them to the `kbuild` file, the same way as is described in the [driver example](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/development/howto/#adding-the-driver-to-the-build)
